File: clients/javascript/src/models/msgQueueConfigureIn.ts
// this file is @generated

export interface MsgQueueConfigureIn {
    topic: string;
    consumerGroup: string;
    retrySchedule?: number[];
    dlqTopic?: string | null;
}

export const MsgQueueConfigureInSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): MsgQueueConfigureIn {
        return {
            topic: object['topic'],
            consumerGroup: object['consumer_group'],
            retrySchedule: object['retry_schedule'],
            dlqTopic: object['dlq_topic'],
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: MsgQueueConfigureIn): any {
        return {
            'topic': self.topic,
            'consumer_group': self.consumerGroup,
            'retry_schedule': self.retrySchedule,
            'dlq_topic': self.dlqTopic,
        };
    }
}

File: z-clients/python/diom/models/msg_queue_configure_in.py
# this file is @generated
import typing as t

from ..internal.base_model import BaseModel


class MsgQueueConfigureIn(BaseModel):
    namespace: str | None = None

    retry_schedule: t.List[int] | None = None

    dlq_topic: str | None = None


class _MsgQueueConfigureIn(BaseModel):
    namespace: str | None = None

    topic: str

    consumer_group: str

    retry_schedule: t.List[int] | None = None

    dlq_topic: str | None = None
